sdk/template/Azure.Template/src/Generated/MultiVersionClient.cs

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Template
+{
+    /// <summary> The MultiVersion service client. </summary>
+    public partial class MultiVersionClient
+    {
+        private static readonly string[] AuthorizationScopes = new string[] { "https://vault.azure.net/.default" };
+        private readonly TokenCredential _tokenCredential;
+        private readonly HttpPipeline _pipeline;
+        private readonly Uri _endpoint;
+        private readonly string _apiVersion;
```

Comment:
why do we store both version enum and string?

Reply:
The string is needed to add api-version to requests, and the enum is needed for comparison of enum values for version checks, e.g. here: https://github.com/Azure/azure-sdk-for-net/pull/30646/files#diff-54256e001b630987030f146f058babafde8e370adcfae19d8c84be16a788e5baR150

My thinking in storing both was that we could keep just the enum and get the string from it (but not vice versa), but that puts a string allocation on a hot path, i.e. request creation.  If we don't consider request creation to be on a hot path, we could store the enum instead, but this would remove an internal field on service-specific client options, which would require significant refactoring across libraries.

That's how I got to this decision, but I'm open to feedback - I'm not wild about this duplication.